gamemodes/terrortown/gamemode/server/sv_player_ext.lua

```diff
@@ -1311,6 +1311,135 @@ function plymeta:SafeDropWeapon(wep, keepSelection)
 	return true
 end
 
+local function TraceAmmoDrop(ply)
+	local pos, ang = ply:GetShootPos(), ply:EyeAngles()
+	local fwd, rgt, up = ang:Forward(), ang:Right(), ang:Up()
+
+	local dir = fwd * 32
+	rgt:Mul(6)
+	up:Mul(-5)
+	dir:Add(rgt)
+	dir:Add(up)
+
+	local tr = util.QuickTrace(pos, dir, ply)
+
+	return tr, pos, dir, fwd
+end
+
+---
+-- Checks if the ammo can be dropped in a safe manner.
+-- @param Weapon wep The weapon's ammo that should be dropped
+-- @return boolean Returns if this weapon's ammo can be dropped
+-- @realm server
+function plymeta:CanSafeDropAmmo(wep)
+	if not IsValid(self) or not (IsValid(wep) and wep.AmmoEnt) then
+		return false
+	end
+
+	local tr = TraceAmmoDrop(self)
+
+	if tr.HitWorld then
+		LANG.Msg(self, "drop_no_room_ammo", nil, MSG_MSTACK_WARN)
+
+		return false
+	end
+
+	return true
+end
+
+---
+-- Called to drop ammo in a safe manner (e.g. preparing and space-check).
+-- @param Weapon wep The weapon that should be referenced when dropping ammo.
+-- @param boolean useClip If set to true the ammo is dropped from the clip, otherwise, from reserve ammo.
+-- @return boolean Returns if ammo is dropped
+-- @realm server
+function plymeta:SafeDropAmmo(wep, useClip)
+	if not self:CanSafeDropAmmo(wep) then
+		return false
+	end
+
+	self:AnimPerformGesture(ACT_GMOD_GESTURE_ITEM_GIVE)
+
+	WEPS.DropNotifiedAmmo(self, wep, useClip)
+
+	return true
+end
+
+---
+-- Called to drop a weapon's ammo. Does no safety checks.
+-- @param Weapon wep The weapon's ammo that should be dropped
+-- @param boolean useClip If set to true the ammo is dropped from the clip, otherwise, from reserve ammo.
+-- @param number amt The quantity of ammo to drop.
+-- @return boolean Returns if this weapon's ammo is dropped.
+-- @realm server
+function plymeta:DropAmmo(wep, useClip, amt)
+	local box = ents.Create(wep.AmmoEnt)
+
+	if not IsValid(box) then return end
+
+	-- most of this goes into computing ammo amount
+	amt = amt or 0
```

Comment:
```suggestion
	amt = amt or 0

	local box = ents.Create(wep.AmmoEnt)

	if not IsValid(box) then return end

	-- most of this goes into computing ammo amount
```

validation of parameters should be on the top of the function, as these represent the default values and should therefore be visible at first glance